content_store_controller.go

```diff
@@ -13,54 +14,69 @@ import (
+	controller.registerWithURLArbiterAndForward(w, r, func(path string, requestBody []byte) {
+		controller.doDraftContentStoreRequest("PUT", strings.Replace(path, "draft-", "", -1), requestBody, w)
+	})
+}
 
-	requestBody, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		renderer.JSON(w, http.StatusInternalServerError, err)
-		return
-	}
+func (controller *ContentStoreController) PutContentStoreRequest(w http.ResponseWriter, r *http.Request) {
+	controller.registerWithURLArbiterAndForward(w, r, func(path string, requestBody []byte) {
+		// TODO: PUT to both content stores concurrently
+		// for now, we ignore the response from draft content store for storing live content, hence `w` is nil
+		controller.doContentStoreRequest("PUT", path, requestBody, w)
+		controller.doDraftContentStoreRequest("PUT", path, requestBody, nil)
+	})
+}
 
-	var contentStoreRequest *ContentStoreRequest
-	if err := json.Unmarshal(requestBody, &contentStoreRequest); err != nil {
-		switch err.(type) {
-		case *json.SyntaxError:
-			renderer.JSON(w, http.StatusBadRequest, err)
-		default:
-			renderer.JSON(w, http.StatusInternalServerError, err)
-		}
-		return
-	}
+func (controller *ContentStoreController) PutPublishIntentRequest(w http.ResponseWriter, r *http.Request) {
+	controller.registerWithURLArbiterAndForward(w, r, func(path string, requestBody []byte) {
+		controller.doContentStoreRequest("PUT", path, requestBody, w)
+	})
+}
 
-	if !cs.registerWithURLArbiter(urlParameters["base_path"], contentStoreRequest.PublishingApp, w) {
-		// errors already written to ResponseWriter
-		return
-	}
+func (controller *ContentStoreController) GetContentStoreRequest(w http.ResponseWriter, r *http.Request) {
+	controller.doContentStoreRequest("GET", r.URL.Path, nil, w)
+}
+
+func (controller *ContentStoreController) DeleteContentStoreRequest(w http.ResponseWriter, r *http.Request) {
+	controller.doContentStoreRequest("DELETE", r.URL.Path, nil, w)
+}
+
+func (controller *ContentStoreController) registerWithURLArbiterAndForward(w http.ResponseWriter, r *http.Request,
+	afterRegister func(path string, requestBody []byte)) {
 
-	cs.doContentStoreRequest("PUT", r.URL.Path, requestBody, w)
+	urlParameters := mux.Vars(r)
+	if requestBody, contentStoreRequest := controller.readRequest(w, r); contentStoreRequest != nil {
+		if !controller.registerWithURLArbiter(urlParameters["base_path"], contentStoreRequest.PublishingApp, w) {
+			return
+		}
+		afterRegister(r.URL.Path, requestBody)
+	}
 }
 
 // Register the given path and publishing app with the URL arbiter.  Returns
 // true on success.  On failure, writes an error to the ResponseWriter, and
 // returns false
-func (cs *ContentStoreController) registerWithURLArbiter(path, publishingApp string, w http.ResponseWriter) bool {
-	urlArbiterResponse, err := cs.arbiter.Register(path, publishingApp)
+func (controller *ContentStoreController) registerWithURLArbiter(path, publishingApp string, w http.ResponseWriter) bool {
```

Comment:
Is it worth calling the `path` parameter `basePath` to make it clear that it's the base path of the item without any prefix components etc.